src/main/initVerticalMod.F90

```diff
@@ -213,41 +215,37 @@ subroutine initVertical(bounds, glc_behavior, snow_depth, thick_wall, thick_roof
     ! Soil layers and interfaces (assumed same for all non-lake patches)
     ! "0" refers to soil surface and "nlevsoi" refers to the bottom of model soil
 
-    if ( soil_layerstruct == '10SL_3.5m' ) then 
-       do j = 1, nlevgrnd
-          zsoi(j) = scalez*(exp(0.5_r8*(j-0.5_r8))-1._r8)    !node depths
-       enddo
-
-       dzsoi(1) = 0.5_r8*(zsoi(1)+zsoi(2))             !thickness b/n two interfaces
-       do j = 2,nlevgrnd-1
-          dzsoi(j)= 0.5_r8*(zsoi(j+1)-zsoi(j-1))
-       enddo
-       dzsoi(nlevgrnd) = zsoi(nlevgrnd)-zsoi(nlevgrnd-1)
-
-       zisoi(0) = 0._r8
-       do j = 1, nlevgrnd-1
-          zisoi(j) = 0.5_r8*(zsoi(j)+zsoi(j+1))         !interface depths
-       enddo
-       zisoi(nlevgrnd) = zsoi(nlevgrnd) + 0.5_r8*dzsoi(nlevgrnd)
+    if (soil_layerstruct_predefined == '10SL_3.5m' .or. soil_layerstruct_predefined == '23SL_3.5m') then
```

Comment:
Do you know if all of the predefined options are covered by at least one system test? If there is an option that is NOT covered by a system test, it would be good to run a one-time system test comparing your branch against master. (i.e., set up a test on master and your branch, change user_nl in both, run them and ensure bit-for-bit; a 5-day SMS test would be sufficient.)

Reply:
All but one, as far as I can tell.
10SL... gets tested in clm4.5 tests
23SL... gets tested in vrtlay tests
20SL... gets tested in clm5 tests
4SL... gets tested in nwp tests
49SL... DOES NOT GET TESTED
I can do that.

Reply:
I tried
./create_test ERI_D_Ld9.ne30_g16.I2000Clm50BgcCruGs.cheyenne_intel.clm-vrtlay
with soil_layerstruct_predefined changed to '49SL_10m'  and got **FAIL**  with an abort in soilbiogeochemlittverttranspmod, which may be revealing a bug here, but I'll need to come back to this later.

I went back and tried the test unchanged (soil_layerstruct_predefined = '23SL_3.5m’) and got roundoff DIFF in three variables from dev050. I hadn't expected diffs given my changes. I guess I need to back out changes affecting 23SL_3.5m and try again...

Reply:
Oops, I was supposed to compare to dev051. Trying again.

Reply:
Since the point of this is just to make sure the soil layer structure is set up identically to before, you could try a non-BGC (i.e., SP) test with 49SL_10m and see if that is bfb. If that still fails, I'd be comfortable enough if you just print out all of the relevant variables in initialization, before (i.e., on master) and after your changes, and verify that these variables (dzsoi, zsoi, nlevsoi, etc.) are identical in the log files (if you haven't already done so).

Reply:
> Oops, I was supposed to compare to dev051. Trying again.

Ok, this PASSed. Now waiting for cheyenne's compute nodes to return to svc...